pyanaconda/ui/gui/spokes/network.py

```diff
@@ -1221,6 +1123,211 @@ def _try_disconnect(obj, callback):
+        treeview.append_column(col)
+
+    @property
+    def selected_ssid(self):
+        ssid = b""
+        selection = self._treeview.get_selection()
+        model, itr = selection.get_selected()
+        if itr:
+            ssid = model[itr][SELECT_WIRELESS_COLUMN_SSID]
+        return ssid
+
+    def _get_strongest_unique_aps(self, access_points):
+        strongest_aps = {}
+        for ap in access_points:
+            if not ap.get_ssid():
+                # non-broadcasting AP. We don't do anything with these
+                continue
+            ssid = ap.get_ssid().get_data()
+            if ssid in strongest_aps:
+                if ap.get_strength() > strongest_aps[ssid].get_strength():
+                    strongest_aps[ssid] = ap
+            else:
+                strongest_aps[ssid] = ap
+
+        return strongest_aps.values()
+
+    # pylint: disable=arguments-differ
+    def refresh(self, device_name):
+        device = self._nm_client.get_device_by_iface(device_name)
+        if not device:
+            log.warnig("device for interface %s not found", device)
+            return
+
+        self._device_name = device_name
+
+        aps = self._get_strongest_unique_aps(device.get_access_points())
+
+        active_ap = device.get_active_access_point()
+        if active_ap:
+            active_ssid = active_ap.get_ssid().get_data()
+        else:
+            active_ssid = b""
+
+        # Update model
+        self._store.clear()
+        for ap in aps:
+            self._add_ap(ap, active_ssid)
+
+    def _add_ap(self, ap, active_ssid):
+        ssid = ap.get_ssid()
+        if not ssid:
+            # get_ssid can return None if AP does not broadcast.
+            return
+
+        ssid = ssid.get_data()
+        if not ssid:
+            return
+
+        mode = ap.get_mode()
+        if not mode:
+            return
+
+        security = self._ap_security(ap)
+
+        ssid_str = NM.utils_ssid_to_utf8(ssid)
+
+        active = ssid == active_ssid
+
+        # the third column is for sorting
+        self._store.append([ssid, ssid_str, ssid_str, ap.get_strength(), mode, security, active])
+
+    def _ap_security(self, ap):
+        ty = NM_AP_SEC_UNKNOWN
+
+        flags = ap.get_flags()
+        wpa_flags = ap.get_wpa_flags()
+        rsn_flags = ap.get_rsn_flags()
+
+        if not flags & NM._80211ApFlags.PRIVACY \
+                and wpa_flags == NM._80211ApSecurityFlags.NONE \
+                and rsn_flags == NM._80211ApSecurityFlags.NONE:
+            ty = NM_AP_SEC_NONE
+        elif flags & NM._80211ApFlags.PRIVACY \
+                and wpa_flags == NM._80211ApSecurityFlags.NONE \
+                and rsn_flags == NM._80211ApSecurityFlags.NONE:
+            ty = NM_AP_SEC_WEP
+        elif not flags & NM._80211ApFlags.PRIVACY \
+                and wpa_flags != NM._80211ApSecurityFlags.NONE \
+                and rsn_flags != NM._80211ApSecurityFlags.NONE:
```

Comment:
Could you please abstract these checks to a methods or variables? It would be much better to read with proper names.

Reply:
I don't think it is worth the effort.